File: java/create_scraper_list.java
import java.io.FileWriter;
import java.io.IOException;
import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;

public class Main {

    public static void main(String[] args) {
        String url = "https://api.minexa.ai/scraper/";
        String apiKey = "YOUR_API_KEY";

        System.out.println("Creating Scraper.. This may take up to 2 minutes");

        JSONObject data = new JSONObject();
        JSONArray urls = new JSONArray();
        urls.add("https://clinicaltrials.gov/search?page=1");

        // Copy paste the data from the website directly without any modification to train the scraper
        // or Write a short description of your required data, for example: "clinical trials results"
        data.put("look_for", """
        NCT06939504
        Not yet recruiting
        New
        A Trial of HRS-9813 Capsule and Tablet in Healthy Subjects
        Conditions
        Pulmonary Fibrosis
        Locations
        Shanghai, Shanghai, China


        NCT06939491
        Recruiting
        New
        Comparison of Ultrasound-guided Electrolysis Therapy vs. Sham Electrolysis in Patients With Patellar Tendinopathy: A Prospective Randomized Study Including MRI and Shear-wave Ultrasound Elastography Imaging
        Conditions
        Patellar Tendinopathy / Jumpers Knee
        Locations
        Bremen, Germany


        NCT06939478
        Not yet recruiting
        New
        AI Powered Mapping Technology for Identifying Arrhythmias
        Conditions
        Arrhythmias, Cardiac
        Locations
        Location not provided


        NCT06939465
        Recruiting
        New
        Esophageal Visceral Hypersensitivity and Hypervigilance in Disorders of Gut-brain Interaction: the Roles of Cognitive-behavioral Therapy
        Conditions
        GERD Without Erosive Esophagitis
        Gut-Brain Disorders
        Locations
        Hualien, Taiwan""");
        data.put("urls", urls);
        data.put("mode", "list");

        String requestBody = data.toString();

        try {
            URL obj = new URL(url);
            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
            con.setRequestMethod("POST");
            con.setRequestProperty("Content-Type", "application/json");
            con.setRequestProperty("api-key", apiKey);

            con.setDoOutput(true);
            OutputStream os = con.getOutputStream();
            os.write(requestBody.getBytes());
            os.flush();
            os.close();

            int responseCode = con.getResponseCode();
            if (responseCode == HttpURLConnection.HTTP_OK) { // success
                // Print the response
                System.out.println("Please confirm container is well located: " + con.getResponseMessage());

                // Read and save the response
                // Here, you can write code to read the response and save it to a file
                // For simplicity, I'm not implementing it here

                System.out.println("Full scraper json saved at: <file_path>");

            } else {
                System.out.println("Error status code " + responseCode + " occurred");
                // Print response message if any
                System.out.println(con.getResponseMessage());
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: php/create_scraper_list.php
<?php
$url = "https://api.minexa.ai/scraper/";
$api_key = "YOUR_API_KEY";

// Your JSON data for the request body
$data = array(

    // Copy paste the data from the website directly without any modification to train the scraper
    // or Write a short description of your required data, for example: "clinical trials study overview"
    "look_for" => <<<EOD
    Study Overview
    Brief Summary
    The gold standard after shoulder resection for tumors is reconstruction by reverse prosthesis and allograft. This is an intervention also performed for more frequent etiologies (revisions of prosthesis, non cancerous humeral bone loss ...).

    The results in these etiologies are good, and do not find any particular mechanical complications (including no osteolysis of the graft). In the case of reconstruction for cancer, the numbers of patients are lower (rare pathologies) and some studies on small numbers found osteolysis of the allograft. The aim of this study is to analyze the presence or not osteolysis in these patients, and to quantify it precisely by scanner measurement (no data yet in the literature).

    Detailed Description
    quantify bone stock of the allograft by scanner measurement in post operative and in 6 month to 1 years after surgery.

    This is a retrospective study, and the scanner was performed routinely every 3 to 6 month, during 2 years, for oncological follow up.

    Official Title
    Results of Proximal Humeral Reconstruction With Allograft Prosthetic Composite After Resection for Tumors
    Conditions
    Shoulder Disease
    Intervention / Treatment
    Procedure: proximal humeral resection for tumor and allograft prosthetic composite reconstruction
    Other Study ID Numbers
    2024PI059,
    EOD,
    // Provide 4 urls corresponding to a similarly structured page but with different data
    "urls" => array(
        "https://clinicaltrials.gov/study/NCT06382792",
        "https://clinicaltrials.gov/study/NCT06382779",
        "https://clinicaltrials.gov/study/NCT06382753",
        "https://clinicaltrials.gov/study/NCT06382727"
    ),

    // uncomment if you need to recrawl the HTML again from scratch by ignoring cached data (like its the first time you scrape it)
    //"reset" => true,

    // Unocomment and set it when manaully detecting your container after first try
    // No need to use for creating scraper for a particular page for the first time
    // "xpath" => "/html/body/div/div/div[3]/div[1]/div[2]",

    // if you want to extract detail data, we advise not to use a simple domain name but try to find pages that are different
    // Use detail if the data is mixed and in a less structured format.
    "mode" => "detail",
);

$headers = array(
    "Content-Type: application/json",
    "api-key: $api_key"
);

echo "Creating scraper.. This may take up to 2 minutes"

// Make the POST request
$ch = curl_init($url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
$response = curl_exec($ch);
$status_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

// Print the response
if ($status_code == 200) {
    $response_data = json_decode($response, true);
    echo "Please confirm container is well located: " . $response_data['response']['web_app'] . "\n";

    // Create and save the file
    $scraper_id = $response_data["response"]["id"];
    $file_path = dirname(__FILE__) . "/scraper_id_" . $scraper_id . ".json";

    // Save the scraper.json
    file_put_contents($file_path, json_encode($response_data, JSON_PRETTY_PRINT));
    echo "Full scraper json saved at: $file_path\n";
} else {
    echo "Error status code $status_code occurred \n";
    try {
        $error_data = json_decode($response, true);
        echo json_encode($error_data, JSON_PRETTY_PRINT) . "\n";
    } catch (Exception $e) {
        echo $e->getMessage() . " in showing error\n";
    }
}
?>
